src/modules/company/controllers/deleteAdmin.js

```diff
@@ -10,28 +19,6 @@ export const deleteAdminFromCompany = async (databasePrisma, req, res) => {
   if (!admin) {
     return res.status(400).send({ message: "Admin id is required" });
   }
-  //Validate to see if a user is logged in
-  const token = req.headers.authorization;
-  let JWT = token;
-  if (!token) {
-    return res.status(401).send({
-      error: "User has to be authenticated to make this request",
-    });
-  } else if (token.includes("Bearer")) {
-    JWT = token.slice(7);
-  }
-  if (JWT === undefined) {
-    return res.status(401).send({
-      message: "No authorization header provided.",
-    });
-  }
-  var verified = await verifyToken(JWT);
-  if (!verified) {
-    return res.status(401).send({
-      message: "Authorization token is not valid.",
-    });
-  }
-
   //Check if company exists
```

Comment:
Much of the code from line 22 to 71 can be removed as this is checked by the middleware.

ultimately the controller should only really deal with the act of removing the admin from the company, as well as any unique logic that is relevant to it, which in this case could be a check that the company has more than one admin so companies can't be left without an admin.

Reply:
@Anclagen I agree. Will have to revise the whole deleteAdmin.js. Will look into it!